### ISSUE TITLE
False-negative `used-before-assignment` after try-except block

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#5524]
Followup to https://github.com/PyCQA/pylint/issues/5500#issuecomment-991695275

```py
def func(var):
    try:
        return 1 / var.some_other_func()
    except AttributeError:
        pass
    except ZeroDivisionError:
        msg = "Devision by 0"

    print(msg)  # should emit 'used-before-assignment'
```

The code above should raise a `used-before-assignment` error since `msg` is not assigned in the `AttributeError` ExceptHandler.

**Comment by jacobtylerwalls** [comment]
Extending this example, if the try block actually defines `msg` instead of returning, since there is not exhaustive error handling (by assigning the name, or raising or exiting), then we could emit a message that the use of the name needs to be guarded by `else`. Maybe that's also `used-before-assigment`, or maybe it's not. 🤔 

Maybe `consider-using-try-else` (or if that's too weasel-worded, maybe `nonexhaustive-error-handling`?)

```python
try:
    a = 1 / n
except LibraryError:
    pass
print(a)  # [nonexhaustive-error-handling]
```

```python
try:
    a = 1 / n
except LibraryError:
    pass
else:
    print(a)  # no message
```

**Comment by jacobtylerwalls** [comment]
We can close #2835 as a duplicate.

**Comment by cdce8p** [comment]
Another idea. I like how pyright names it. Maybe `possibly-unbound`?
```
"a" is possibly unbound
```

> Maybe `consider-using-try-else`

We shouldn't really "force" users to add an `else` case. There are just to many alternatives. You could for example assign a default value before you enter the `try` block.

**Comment by DanielNoord** [comment]
Some of the discussion in this issue is being moved to #2835. The rest is being closed with #5674.
